src/index.ts

```diff
@@ -415,6 +432,87 @@ export class FlashbotsBundleProvider extends providers.JsonRpcProvider {
+   * */
+
+  public async publish(
+    bundle: FlashbotsBundle,
+    validatorPublicKey: string,
+    signer: Signer,
+    carrierTx: TransactionRequest
+  ): Promise<TransactionResponse> {
+    //RLP-serialize the given bundle
+    const serializedBundle = this.rlpSerializeBundle(bundle)
+
+    //Encrypt the encoded bundle with the passed validator pub_key
+    const encryptedBundle = encrypt(validatorPublicKey, Buffer.from(serializedBundle))
+
+    //Check if carrier_tx has minimum params, populate with defaults if not
+    /*
+     The following statement is intended to be used in order to support any type of incomplete TransactionRequest
+     received, populating it with default values if any one is missing
+     */
+    await this.prepareCarrierTransaction(carrierTx, signer)
+
+    //Populate carrier_tx.data as : carrier_tx.data = MEV_Prefix | validator pub_key | Encrypt(validator pub_key, serialized bundle)
+    const mevPrefix = `0123` //this is a placeholder!
+
+    let payload = `0x`
+    payload += mevPrefix
+    payload += validatorPublicKey
+    payload += encryptedBundle.toString('hex')
+
+    carrierTx.data = payload
+
+    //Sign the transaction received as param with passed signer
+    const signedTx = await signer.signTransaction(carrierTx)
+
+    //Propagate carrier_tx into the public mempool and return Promise<TransactionResponse> for the carrier_tx
+    return this.genericProvider.sendTransaction(signedTx)
+  }
+
+  /**
+   * A private method to encode a FlashbotsBundle following the RLP serialization standard
+   * @param bundle the FlashbotsBundle instance to be serialized
+   * @return string the rlp encoded bundle
+   * @private
+   */
+  private rlpSerializeBundle(bundle: FlashbotsBundle): string {
+    const stringifiedBundle = JSON.stringify(bundle)
+    const bytesBundle = Buffer.from(stringifiedBundle)
+    return encode(bytesBundle)
+  }
+
+  /**
+   * A private method to populate {@param carrier}'s missing fields with default values
+   * @param carrier an instance of TransactionRequest which will be the tx containing the full payload in its data field
+   * @param signer an instance of Signer, which is used to retrieve the correct nonce for the carrier tx, should it be missing
+   * @private
+   */
+  private async prepareCarrierTransaction(carrier: TransactionRequest, signer: Signer) {
+    if (!('to' in carrier)) carrier.to = '0x0000000000000000000000000000000000000000'
+    if (!('value' in carrier)) carrier.value = 0
+    if (!('gasPrice' in carrier)) carrier.gasPrice = await this.genericProvider.getGasPrice()
+    if (!('nonce' in carrier)) carrier.nonce = await this.genericProvider.getTransactionCount(await signer.getAddress())
+    if (!('gasLimit' in carrier)) carrier.gasLimit = 210000 //Default value of 210k gasLimit, which is 10x of a simple transfers gasLimit
+
+    //TODO How can I calculate the exact gasLimit?? --> Found different resource that said that G_txdatanonzero is either 16 or 68
```

Comment:
can't calculate it exactly if you aren't willing to simulate it, i recommend looking into the json rpc method estimateGas. i wouldn't just set it to 210000, that is a pretty random number

Reply:
Can it not be calculated as a function of the data field's length? as the carrier is a simple transfer tx and will not be making any contract calls